chia/full_node/full_node.py

```diff
@@ -214,160 +215,162 @@ async def manage(self) -> AsyncIterator[None]:
         self._add_transaction_semaphore = asyncio.Semaphore(200)
 
         sql_log_path: Optional[Path] = None
-        if self.config.get("log_sqlite_cmds", False):
-            sql_log_path = path_from_root(self.root_path, "log/sql.log")
-            self.log.info(f"logging SQL commands to {sql_log_path}")
-
-        # create the store (db) and full node instance
-        # TODO: is this standardized and thus able to be handled by DBWrapper2?
-        async with manage_connection(self.db_path, log_path=sql_log_path, name="version_check") as db_connection:
-            db_version = await lookup_db_version(db_connection)
+        with contextlib.ExitStack() as exit_stack:
+            sql_log_file: Optional[TextIO] = None
+            if self.config.get("log_sqlite_cmds", False):
+                sql_log_path = path_from_root(self.root_path, "log/sql.log")
+                self.log.info(f"logging SQL commands to {sql_log_path}")
+                sql_log_file = exit_stack.enter_context(sql_log_path.open("a", encoding="utf-8"))
+
+            # create the store (db) and full node instance
+            # TODO: is this standardized and thus able to be handled by DBWrapper2?
+            async with manage_connection(self.db_path, log_file=sql_log_file, name="version_check") as db_connection:
+                db_version = await lookup_db_version(db_connection)
+
```

Comment:
the exit stack will destruct here, and close the `sql_log_file`, right?
it's not obvious how this will work. By the time the DBWrapper logs to the file, won't it already be closed?

Reply:
The log aspect of the interface isn't super great/consistent and we could think it through a bit.  Including this PR, the `manage_connection()` manager takes an open file.  Yes, that file will be closed at the end of this exit stack.  The `DBWrapper2.managed()` manager takes a path, not a file.  It then opens that file and uses it for all the connections.  So, we open the log file twice in this case.  Once when checking the db version, and again when actually setting up the wrapper.

Reply:
I see